### PR TITLE
hyperscan: skip compilation on unused libs

### PR DESCRIPTION
**Pull request by zhxie**
Signed-off-by: Xie Zhihao <zhihao.xie@intel.com>

Commit Message: hyperscan: skip compilation on unused libs
Additional Description:
Risk Level: Low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

File: contrib/hyperscan/matching/input_matchers/source/BUILD [hunk 1]
```diff
@@ -26,6 +26,7 @@ envoy_cmake(
         "BUILD_AVX512": "on",
         "BUILD_AVX512VBMI": "on",
         "BUILD_EXAMPLES": "off",
+        "CMAKE_BUILD_TYPE": "Release",
```
Comment (phlax) [inline contrib/hyperscan/matching/input_matchers/source/BUILD:29]: i think you will need to add the hack to override the default here - add:

```starlark
default_cache_entries = {}
```
to the cmake rule

**Review by phlax — COMMENTED** [review]
thanks for taking a look at this @zhxie - really appreciated

File: contrib/hyperscan/matching/input_matchers/source/BUILD (hunk 1 above)
Comment (zhxie) [reply contrib/hyperscan/matching/input_matchers/source/BUILD:29]: Yeah, forgot to clean up default entries.

**Comment by zhxie** [comment]
There is not too much unused libs to remove.

In fact, Hyperscan does not have many targets to compile, but it compiles a separate copy of the code for each platform (SSE4, AVX, AVX2...) in order to optimize performance on different platforms.

**Review by phlax — APPROVED** [review]
thanks for following up on this @zhxie, lgtm 

locally my comparison (test of 1) timings:

before:

```
INFO: Elapsed time: 399.542s, Critical Path: 399.16s

```

after

```
INFO: Elapsed time: 347.085s, Critical Path: 338.38s
```

as this definitely does build on all cores -  i think we should try and figure out how to get the ci machines to build it locally rather than compiling with RBE

**Pull request merged**